docs/module-sources.rst

```diff
@@ -0,0 +1,562 @@
+
+File sources
+------------
+
+These are sources of ``type: file``. They are copied into the source
+directory without any modifications.
+
+Any arbitrary file type can be specified here along with
+``buildsystem: simple`` to manually modify, build or install it
+within ``build-commands``. This can be used in place of ``type: archive``
+to manually extract an archive, uncompress squashfs filesystems
+(like snaps) etc. provided the tool being used is made available inside
+the sandbox.
+
+A typical example is:
+
+.. code-block:: yaml
+
+  modules:
+    - name: foo
+      buildsystem: simple
+      build-commands:
+       - install -Dm0644 org.example.foo.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo/
+       - bsdtar -Oxf foo.deb 'data.tar.xz' | bsdtar -xf -
+       - mv opt/ ${FLATPAK_DEST}/
+       - ln -s ${FLATPAK_DEST}/opt/executable ${FLATPAK_DEST}/bin/executable
+      sources:
+        - type: file
+          dest-filename: foo.deb
+          url: https://example.org/foo-v1.0.deb
+          sha256: <sha256 checksum of foo.deb>
+        - type: file
+          path: org.example.foo.metainfo.xml
+
+    - name: unsquashfs
+      buildsystem: simple
+      build-commands:
+       - XZ_SUPPORT=1 make -C squashfs-tools -j ${FLATPAK_BUILDER_N_JOBS} unsquashfs
+       - install -Dpm755 -t "${FLATPAK_DEST}/bin" squashfs-tools/unsquashfs
+      sources:
+        - type: git
+          url: https://github.com/plougher/squashfs-tools.git
+          tag: 4.6.1
+          commit: d8cb82d9840330f9344ec37b992595b5d7b44184
+
+    - name: bar
+      buildsystem: simple
+      build-commands:
+       - unsquashfs -dest bar -quiet -no-progress squashed.snap
+       - mv bar ${FLATPAK_DEST}
+       - ln -s ${FLATPAK_DEST}/bar/executable ${FLATPAK_DEST}/bin/executable
+      sources:
+        - type: file
+          dest-filename: squashed.snap
+          url: https://example.org/squashed-1.0.snap
+          sha256: <sha256 checksum of squashed-1.0.snap>
+
+
+``dest-filename`` is typically used with this source type to make the
+filename `predictable` for the commands in the ``build-commands`` array.
+
+Patch sources
+-------------
+
+These are sources of ``type: patch`` used with patchfiles. The typical
+usage is to create patchfiles and place them somewhere relative to the
+manifest root. The ``path`` property is the location of the file relative
+to the manifest.
+
+.. code-block:: yaml
+
+  modules:
+    - name: foo
+      buildsystem: cmake-ninja
+      builddir: true
+      config-opts:
+      - ...
+    sources:
+      - type: git
+        url: https://example.org/foo.git
+        tag: v1.0.0
+        commit: 63c01f14391aef7d7691c7c63a610d47512147ed
+      - type: patch
+        path: patches/test-patch.patch
+
+Multiple patches can be specified with ``paths``:
+
+.. code-block:: yaml
+
+  - type: patch
+    paths:
+      - patches/test1-patch.patch
+      - patches/test2-patch.patch
+
+By default, ``patch -p 1`` is used, but it can be adjusted with
```

Comment:
We should mention that the default are suited for the default format that git generates.